src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,553 @@
+    consistency in all arithmetic operations.
+    """
+
+    def __init__(self, real_value, unit_str):
+        self._value = float(real_value)
+        self._unit = Unit(unit_str)
+        self._dimension = Dimension(unit_str)
+
+    @property
+    def value(self):
+        return self._value
+
+    @property
+    def unit(self):
+        return self._unit.user_unit
+
+    @value.setter
+    def value(self, val):
+        self._value = val
+
+    @property
+    def dimension(self):
+        return self._dimension
+
+    @property
+    def si_value(self):
+        return self._unit.si_factor * self._value
+
+    @property
+    def si_unit(self):
+        return self._unit.si_unit
+
+    def is_dimension_less(self):
+        return all([value == 0 for value in self.get_dimensions_list()])
+
+    def get_dimensions_list(self):
+        dims = self._dimension.get_dimensions_dict()
+        return [
+            dims[_UnitsTable.dimension_order[key]]
+            for key in _UnitsTable.dimension_order.keys()
+        ]
+
+    def to(self, to_unit_str):
+        temp_quantity = Quantity(1, to_unit_str)
+
+        curr_unit_dim_obj = self.dimension
+        to_unit_dim_obj = temp_quantity.dimension
+
+        if (
+            curr_unit_dim_obj.get_dimensions_dict()
+            != to_unit_dim_obj.get_dimensions_dict()
+        ):
+            raise ValueError(
+                f"Incompatible conversion from {self.unit} : to {to_unit_str}"
+            )
+        else:
+            pass
+
+        curr_unit_obj = self._unit
+        to_unit_obj = temp_quantity._unit
+        temp_quantity.value = (
+            curr_unit_obj.si_factor / to_unit_obj.si_factor
+        ) * self.value
+
+        return temp_quantity
+
+    def _get_si_unit(self, other, func):
+        curr_dim = self.get_dimensions_list()
+        other_dim = other.get_dimensions_list()
+        temp_dim = [func(curr_dim[i], other_dim[i]) for i in range(len(other_dim))]
+        temp_unit = get_si_unit_from_dim(temp_dim)
+        return temp_unit
+
+    def __pow__(self, exponent):
+        new_dims = list(
+            map(lambda x: x * exponent if x != 0 else x, self.get_dimensions_list())
+        )
+        new_si_unit = get_si_unit_from_dim(new_dims)
+        new_si_value = pow(self.si_value, exponent)
+        return Quantity(new_si_value, new_si_unit)
+
+    def __float__(self):
+        return float(self.si_value)
+
+    def __str__(self):
+        return f'({self.value}, "{self.unit}")'
+
+    def __repr__(self):
+        return f'(Quantity ({self.value}, "{self.unit}"))'
+
+    def __mul__(self, other):
+        if isinstance(other, Quantity):
+            temp_value = self.si_value * other.si_value
+            temp_unit = self._get_si_unit(other, lambda x, y: x + y)
+            return Quantity(temp_value, temp_unit)
+        elif isinstance(other, int) or isinstance(other, float):
+            temp = Quantity(self.si_value * other, self.si_unit)
```

Comment:
Shouldn't we retain the units in case of multiplication with scalar?

Reply:
I feel + and - should retain the unit of the left operand.
Multiplication with scalar should retain the unit.
For Quantity(1, 'g')*Quantity(1, 'm'), my expectation is Quantity(1, 'g m')

Reply:
@h-krishnan @seanpearsonuk @dnwillia-work @mkundu1 

I have noted this points. In previous version using `pint` and in current version on `C++` side we are converting all quantities to SI before performing any mathematical operations, because in any version whether it is 'pint' based or `C++`,  we are doing this for consistency. So may be we can have discussion about this and then we can decide.

Thank you.